### PR TITLE
COP-10177 target screen

### PR DESCRIPTION
**Pull request by charles474**
This PR adds the new design for the target capture screen as per the prototype.

There is still a fair amount of work to be done but this allows us to test some of the tickets that make up this target screen.

> Note: There are still `TODOs` littered all over the place which will be resolved once work is complete.

### CODE REVIEW

File: client/src/utils/TabSelectedContext.test.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { TaskSelectedTabProvider } from './TabSelectedContext';
+
+describe('TabSelectedContext', () => {
+  it('can render components without crashing', async () => {
+    const wrapper = mount(
+      <TaskSelectedTabProvider>
+        <div>Hello</div>
+      </TaskSelectedTabProvider>
+    );
+    expect(wrapper).toBeDefined();
```
Comment (michaelruocco) [inline client/src/utils/TabSelectedContext.test.jsx:12]: might just be my ignorance but what is this actually testing?

File: client/src/pages/targets/__components__/TargetsListPage/TargetListCardRow.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import React from 'react';
```
Comment (michaelruocco) [inline client/src/pages/targets/__components__/TargetsListPage/TargetListCardRow.jsx:1]: I am okay if we do it later, but I don't see any tests for this component?

File: client/src/pages/targets/__tests__/__snapshots__/Tabs.test.jsx.snap [hunk 3]
```diff
@@ -0,0 +1,538 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (michaelruocco) [inline client/src/pages/targets/__tests__/__snapshots__/Tabs.test.jsx.snap:1]: what are these .snap files for? I don't think I've seen them used before?

File: client/src/pages/targets/utils.js [hunk 4]
```diff
@@ -18,8 +18,8 @@ export const formatCategory = (category) => {
   return category.slice(-1);
 };
 
-export const formatDate = (date) => {
-  return dayjs(date).format('DD MM YYYY HH:mm');
+export const formatDate = (date, withFormat = 'DD MM YYYY HH:mm') => {
```
Comment (michaelruocco) [inline client/src/pages/targets/utils.js:21]: should be easy enough to add a test for this one I think?

Comment (charles474) [reply client/src/pages/targets/utils.js:21]: the snaps are basically screenshots of the page, similar to when you do `ReactDOMServer.renderToSting(...)`

Comment (charles474) [reply client/src/pages/targets/utils.js:21]: just saved having to dump a whole load of HTML in there

File: client/src/utils/TabSelectedContext.test.jsx (hunk 1 above)
Comment (charles474) [reply client/src/utils/TabSelectedContext.test.jsx:12]: just testing that the page does get rendered and doesn't throw an error, equally i can always just test for the presence of the `hello` string.

File: client/src/pages/targets/__tests__/__snapshots__/Tabs.test.jsx.snap (hunk 3 above)
Comment (charles474) [reply client/src/pages/targets/__tests__/__snapshots__/Tabs.test.jsx.snap:1]: same response as here
https://github.com/UKHomeOffice/cop-ui/pull/309#discussion_r822448886

File: client/src/pages/targets/__components__/TargetsListPage/TargetListCardRow.jsx (hunk 2 above)
Comment (michaelruocco) [reply client/src/pages/targets/__components__/TargetsListPage/TargetListCardRow.jsx:1]: if its covered thats probably okay, just looked like a fair bit of functionality in there, but if you are happy its properly tested thats cool.

**Review by michaelruocco — APPROVED** [review]
LGTM

**Pull request merged**